src/components/DatasetCreateView.jsx

```diff
@@ -119,9 +115,17 @@ export class DatasetCreateView extends React.PureComponent {
   }
 
   getStudyOptions(studies) {
-    const studiesList = [];
-    studies.studies.map(study => studiesList.push({ value: study.id, label: study.name }));
-    return studiesList;
+    const studyOptions = [];
+    studies.studies.map(study => studyOptions.push({ value: study.id, label: study.name }));
+    return studyOptions;
+  }
+
+  getAssetOptions(study) {
+    const assetOptions = [];
+    if (study && study.schema && study.schema.assets) {
+      study.schema.assets.map(asset => assetOptions.push({ value: asset.name, label: asset.name }));
```

Comment:
`map` should only be used to create a new array. It feels strange to use it to mutate something like `assetOptions` here instead of taking a more functional approach. Either `forEach` should be used here, or we should switch assetOptions to use `let` above and then have this line be
```javascript
assetOptions = study.schema.assets.map(asset => ({ value: asset.name, label: asset.name }));
```
(I prefer using `map`)